myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+        # are handled in dudt_i, F_i and G_i; time is excluded from the output;
+        # pace is made Vc.
+        time = model.time()
+        pace = model.binding('pace')
+        special_vars = set(
+            [time]
+            + [v.rhs().var() for v in model.states()]
+            + [v.lhs().var() for v in model.states()]
+        )
+        if pace is not None:
+            special_vars.add(pace)
+
+        # Add metadata
+        export_lines.append('/*')
+        export_lines.append('This file was generated by Myokit.')
+        if model.meta:
+            export_lines.append('')
+            for key, val in sorted(model.meta.items()):
+                export_lines.append(f'{key}: {val}')
+            export_lines.append('')
+        export_lines.append('*/')
+        export_lines.append('')
+
+        # Add empty input parameter list
+        export_lines.append('/* Input parameters */')
+        export_lines.append('in = [ ]')
+        export_lines.append('')
+
+        # Add placeholder protocol
+        if pace is not None:
+            export_lines.append('/* Voltage protocol [mV] */')
+            export_lines.append(
+                'Vc { -80 + 120 * heaviside(t-500) - 80 * heaviside(t-1000) }'
+            )
+            export_lines.append('')
+
+        # Add constants
+        const_vars = (
+            set(model.variables(const=True, deep=True, state=False))
+            - special_vars
+        )
+        for component_label, eq_list in sorted_eqs.items():
+            const_eqs = [
+                eq for eq in eq_list.equations() if eq.lhs.var() in const_vars
+            ]
+            if const_eqs:
+                export_lines.append(f'/* Constants: {component_label} */')
+                for eq in const_eqs:
+                    v = eq.lhs.var()
+                    lhs = var_name(v)
+                    rhs = e.ex(eq.rhs)
+                    qname = v.qname()
+                    unit = '' if v.unit() is None else f' {v.unit()}'
+                    export_lines.append(
+                        f'{lhs} {{ {rhs} }} /* {qname}{unit} */'
+                    )
+                export_lines.append('')
+
+        # Add initial conditions `u_i`
+        export_lines.append('/* Initial conditions */')
+        export_lines.append('u_i {')
+        for v in model.states():
+            lhs = var_name(v)
+            rhs = myokit.float.str(v.initial_value(as_float=True))
+            qname = v.qname()
+            unit = '' if v.unit() is None else f' {v.unit()}'
+            export_lines.append(f'{tab}{lhs} = {rhs}, /* {qname}{unit} */')
+        export_lines.append('}')
+        export_lines.append('')
+
+        # Add initial conditions `dudt_i`
+        export_lines.append('dudt_i {')
+        for v in model.states():
+            lhs = var_name(v.rhs())  # Use intermediary dot_x instead of dot(x)
+            rhs = myokit.float.str(v.initial_value(as_float=True))
```

Comment:
I don't quite get what's happening here. What values is this bit of code defining?

Reply:
It's adding this part from ([diffeq-js](https://www.npmjs.com/package/@martinjrobins/diffeq-js?activeTab=readme)):

```
dudt_i {
  dxdt = 1,
  dydt = 0,
  dzdt = 0,
}
```
It is useful because we can put all non-state variables after this and before `F`, otherwise you get errors like `cannot find variable dxdt` if something depends on `dxdt`.

Should probably set all those to 0.

Reply:
Initialises to zero in 1a212493.